### ISSUE TITLE
project specific options from native file are ignored

### ISSUE THREAD

**Issue by SSE4** [mesonbuild/meson#7699]
**Describe the bug**
according to the [meson documentation](https://mesonbuild.com/Machine-files.html#project-specific-options), native files may specify project-specific options. however, it appears that meson ignores them. 

**To Reproduce**
**meson.build**:
```
project('tutorial', 'cpp')
if get_option('opt')
    library('xx', 'xx.cpp')
else
    library('yy', 'yy.cpp')
endif
```
**meson_options.txt**:
```
option('opt', type: 'boolean')
```
**native.ini**:
```
[project options]
opt = false
```
source files **xx.cpp** and **yy.cpp** don't exist, but it's not important to have them.

running the following command:
```
meson --native-file=native.ini build .
```
results in output:
```
meson.build:3:4: ERROR: File xx.cpp does not exist.
```
however, explicitly specifying option on command line:
```
meson --native-file=native.ini -Dopt=false build .
```
results in correct output:
```
meson.build:5:4: ERROR: File yy.cpp does not exist.
```
same result without native file:
```
meson -Dopt=false build .
```

**Expected behavior**
I'd expect `opt` option specified in native file to be respected by meson, thus **yy** library should be selected for the build.

**system parameters**
* Is this a [cross build](https://mesonbuild.com/Cross-compilation.html) or just a plain native build (for the same computer)?
nope, plain build
* what operating system (e.g. MacOS Catalina, Windows 10, CentOS 8.0, Ubuntu 18.04, etc.)
Windows 10
* what Python version are you using e.g. 3.8.0
Python 3.8.1
* what `meson --version`
0.55.1
* what `ninja --version` if it's a Ninja build
1.8.2

**Comment by dcbaker** [comment]
This feature will be new in meson 0.56. you need meson from git currently to use this

**Comment by xclaesse** [comment]
Why is that doc online already? I thought the website is updated only after release.

**Comment by dcbaker** [comment]
That's a good question. @jpakkane?

**Comment by dcbaker** [comment]
Reopening so we can track why docs are updated to unreleased versions.

**Comment by SSE4** [comment]
@dcbaker btw, is there an ETA for 0.56 release?

**Comment by dcbaker** [comment]
@SSE4 no, we've never really stuck to any timetable. It's usually after all the major regressions are fixed in the current release and jpakkane feels enough time had passed/features are in
